Blog post for enhanced non-OpenAI model support

marklysze opened the pull request:
PR for blog post for the enhanced support of non-OpenAI models through new client classes. Includes Anthropic, Mistral.AI, Together.AI so far.

## Related issue number

See Roadmap issue #2946

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

olgavrou commented on website/blog/2024-06-24-AltModels-Classes/index.mdx:
```diff
@@ -0,0 +1,102 @@
+**NOTE: To integrate this setup into GroupChat, follow the [tutorial](https://microsoft.github.io/autogen/docs/notebooks/agentchat_groupchat) with the same config as above.**
+
+
+## Function Calls
```
is more text going to be added here? if not they should be removed

Hk669 replied:
Yes, the function call tutorial will go in.

olgavrou reviewed (approved):
LGTM when comment is resolved

marklysze commented:
Thanks @olgavrou, I'll put the function calling in and would like to write some text around using these alternative models including some tips/parameters that are useful to tweak and work with. That will take a day or two to do. Hope that's okay.

marklysze commented:
I've updated with the upfront text. Still to do is the function calling example.

marklysze commented:
Thanks @olgavrou and @qingyun-wu, I've added the function calling to the blog post. Have also added a separate notebook for more function calling between different providers (separated it as it was very long).

Let us know what you think and if we can improve it.

Hk669 commented on website/blog/2024-06-24-AltModels-Classes/index.mdx:
```diff
@@ -0,0 +1,392 @@
+For an interesting chess game between Anthropic's Sonnet and Mistral's Mixtral, we've put together a sample notebook that highlights some of the tips and tricks for working with non-OpenAI LLMs. [See the notebook here](https://microsoft.github.io/autogen/docs/notebooks/agentchat_nested_chats_chess_altmodels).
+
+
+### Acknowledgements
+
+This blogpost has been revised with the help of ...
```
@marklysze if there are none, we can remove this section.

marklysze replied:
Thanks @Hk669, I left it in there while it gets reviewed and anyone assisting with the review can go in there (as per the last blog), but if no changes required then, yep, we can remove :).

marklysze replied:
Okay, I've removed for now... will add it in if anyone has edits.

sonichi replied:
The section is not removed yet.

marklysze replied:
Ah, had it ready but didn't fully commit. Done now and I'll bring it up to main as well.

marklysze replied:
Thanks, all done.

marklysze commented:
Thanks for approving @olgavrou!

@qingyun-wu, let us know if you would like anyone else to review or if you want any changes. Groq has been added since it is now merged.

Hk669 commented:
@olgavrou @qingyun-wu are we waiting for all the non-OpenAI models to be merged? can you please review this blog and let us know if you would like anyone else to review or if you want any changes.
